src/coreclr/inc/corinfo.h

```diff
@@ -1685,7 +1685,7 @@ enum CORINFO_FIELD_ACCESSOR
     CORINFO_FIELD_STATIC_ADDR_HELPER,       // static field accessed using address-of helper (argument is FieldDesc *)
     CORINFO_FIELD_STATIC_TLS,               // unmanaged TLS access
     CORINFO_FIELD_STATIC_READYTORUN_HELPER, // static field access using a runtime lookup helper
-
+    CORINFO_FIELD_STATIC_DATASEGMENT,       // static field access from the data segment
```

Comment:
Naming nit: other `CORINFO_FIELD_STATIC` flags describe what kind of "static base" do we have. `DATASEGMENT` doesn't really tell one much. Something like `RELOCATABLE` could perhaps be better...

But best of all would be to update the documentation above to define what this value means.